### ISSUE TITLE
Stop Typing Activity

### ISSUE THREAD

**Issue by JasonJShuler** [Microsoft/BotFramework-Services#14]
How do we send a "stop typing" activity?
If it is not possible, what will it take to make it possible?

This is particularly necessary when the framework is being used for live agent escalation.

**Comment by nyxsys** [comment]
Can you clarify what the specific functionality looks like that you would like to see implemented, and if there is a bug preventing that? To note, if it's "clear the typing indicator when the next message is recieved" this [can be done in webchat](https://stackoverflow.com/questions/48207159/how-to-send-is-typing-notifications-until-response-task-completed) but the functionality might not match if you are using a different channel.

**Comment by JasonJShuler** [comment]
When interacting with a human via facebook messenger or most live chats I have encountered, the typing indicator tends to track whether the other party is actively typing or not. You will see it start and stop if the person at the other end begins typing something then (for example) switches to another window.

The request here is simply to support the ability to notify channels that typing has stopped without having to send a message.

I assume we could work around this on a per-channel basis using channel data - I have seen examples of how it is done via facebook, and I can imagine how we might use the backchannel with the webchat control. But this really seems like functionality that would be better implemented as part of the framework.

**Comment by JasonJShuler** [comment]
It looks like the webchat control shows typing for 3 seconds after getting a typing activity - so the idea is that we need to regularly send typing activities to keep the typing indicator showing. I misunderstood how it works - I thought it would show the typing indicator until it received another activity.

Is the behavior documented anywhere? I might have missed it.

**Comment by nyxsys** [comment]
Looking to find a proper doc for v4, but it is mentioned in the middleware itself [here](https://github.com/Microsoft/botbuilder-js/blob/0313ec3e95a13ba9e5c3dd77a1b427506fbec07b/libraries/botbuilder-core/src/showTypingMiddleware.ts#L13).  As for v3, it is right [here ](https://docs.botframework.com/en-us/node/builder/chat-reference/classes/_botbuilder_d_.middleware.html )in the api docs.

**Comment by dandriscoll** [comment]
Yes, the Typing indicator is ephemeral and intended to last 3 seconds. Most channels standardized on 3 seconds before we built this protocol. See requirement A6000 here: https://github.com/Microsoft/BotBuilder/blob/master/specs/botframework-activity/botframework-activity.md
